test/integration/infrastructure/infrastructure_test.go

```diff
@@ -0,0 +1,825 @@
+	})
+
+	Context("with availabilitySet cluster creating new vNet", func() {
+		AfterEach(func() {
+			framework.RunCleanupActions()
+		})
+
+		It("should successfully create and delete", func() {
+			providerConfig := newInfrastructureConfig(nil, nil, false, false)
+
+			namespace, err := generateName()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = runTest(ctx, logger, c, clientSet, namespace, providerConfig, decoder)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("with availabilitySet cluster using existing vNet and existing Identity", func() {
+		AfterEach(func() {
+			framework.RunCleanupActions()
+		})
+
+		It("should successfully create and delete", func() {
+			foreignName, err := generateName()
+			Expect(err).ToNot(HaveOccurred())
+
+			foreignGroupName := foreignName
+			foreignVNetName := foreignName
+			foreignIdName := foreignName
+
+			Expect(prepareNewResourceGroup(ctx, logger, clientSet, foreignGroupName, *region)).To(Succeed())
+			Expect(prepareNewVNet(ctx, logger, clientSet, foreignGroupName, foreignVNetName, *region, VNetCIDR)).To(Succeed())
+			Expect(prepareNewIdentity(ctx, logger, clientSet, foreignGroupName, foreignIdName, *region)).To(Succeed())
+
+			var cleanupHandle framework.CleanupActionHandle
+			cleanupHandle = framework.AddCleanupAction(func() {
+				By("foreign ResourceGroup teardown")
+				err := teardownResourceGroup(ctx, clientSet, foreignGroupName)
+				Expect(err).ToNot(HaveOccurred())
+
+				framework.RemoveCleanupAction(cleanupHandle)
+			})
+
+			vnetConfig := &azurev1alpha1.VNet{
+				Name:          pointer.StringPtr(foreignVNetName),
+				ResourceGroup: pointer.StringPtr(foreignGroupName),
+			}
+			idConfig := &azurev1alpha1.IdentityConfig{
+				Name:          foreignIdName,
+				ResourceGroup: foreignGroupName,
+			}
+			providerConfig := newInfrastructureConfig(vnetConfig, idConfig, false, false)
+			providerConfig.Identity = idConfig
+
+			namespace, err := generateName()
+			Expect(err).ToNot(HaveOccurred())
+			err = runTest(ctx, logger, c, clientSet, namespace, providerConfig, decoder)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("with zonal cluster creating new vNet and NatGateway", func() {
+		AfterEach(func() {
+			framework.RunCleanupActions()
+		})
+
+		It("should successfully create and delete", func() {
+			providerConfig := newInfrastructureConfig(nil, nil, true, true)
+
+			namespace, err := generateName()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = runTest(ctx, logger, c, clientSet, namespace, providerConfig, decoder)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("with zonal cluster using existing vNet and NatGateway", func() {
+		AfterEach(func() {
+			framework.RunCleanupActions()
+		})
+
+		It("should successfully create and delete", func() {
+			foreignName, err := generateName()
+			Expect(err).ToNot(HaveOccurred())
+
+			foreignGroupName := foreignName
+			foreignVNetName := foreignName
+			foreignIdName := foreignName
```

Comment:
Minor comment: We usually use `var ()` blocks when we declare multiple variables in a row. No necessity to change it now, just wanted to make you aware :)